### ISSUE TITLE
Contest scoring mechanism 

### ISSUE THREAD

**Issue by wubcrow** [young-amateurs-rc/contest-manager#5]
This should be done after #4 is figured out. We need to implement some sort of system that allows us to dynamically specify individual score calculation rules. After we finish this create a new issue that allows us to make pretty graphs and tables to show results.

**Comment by wubcrow** [comment]
Working towards this: https://github.com/young-amateurs-rc/contest-manager/issues/4#issuecomment-447598437

Currently consensus is to have plugins in python form. A good step would be to make a class for further implementations to inherit from.

**Comment by galengold** [comment]
thoughts on scoring:
1. Load all logs into the database
2. cross-reference logs to determine valid QSOs
    - If no log received for dx stn, assume valid
    - If log received for dx stn: valid iff de stn's and dx stn's logs match exchanges
3. use scoring plugin to determine score for valid QSOs

**Comment by galengold** [comment]
columns I need in results table:
- id
- foreign key to contest_id
    - so we can get info about the contest the score is for
- foreign key to upload_id
    - so we can get info about who scored the score
- score
- maybe (because they might not work for every contest):
    - number of qsos
    - number of mults

**Comment by thxo** [comment]
@galengold Better QSY #15 for that since the plugin isn't responsible for actually writing anything into the database.
